src/mpid/ch4/src/ch4r_request.h

```diff
@@ -31,9 +31,8 @@ MPL_STATIC_INLINE_PREFIX MPIR_Request *MPIDIG_request_create(MPIR_Request_kind_t
                                        (void **) &MPIDIG_REQUEST(req, req));
     MPIR_Assert(MPIDIG_REQUEST(req, req));
     MPIDIG_REQUEST(req, req->status) = 0;
-    /* init the request as ready for data as this is the common case, CH4 will should set them to
-     * false when needed */
-    MPIDIG_REQUEST(req, recv_ready) = true;
+    MPIDIG_REQUEST(req, req->recv_async).data_copy_cb = NULL;
+    MPIDIG_REQUEST(req, req->recv_async).recv_type = MPIDIG_RECV_NONE;
```

Comment:
It's probably a good idea to add a wrapper function `MPIDIG_recv_reset` (unfortunately `_init` is taken) so that all `recv_async` code is in a single place.

Reply:
OK